.github/CONTRIBUTING.md

```diff
@@ -232,21 +232,40 @@ python -m pytest -v tests/trainer/test_trainer_cli.py::test_default_args
 
 We welcome any useful contribution! For your convenience here's a recommended workflow:
 
-0. Think about what you want to do - fix a bug, repair docs, etc. If you want to implement a new feature or enhance an existing one, start by opening a GitHub issue to explain the feature and the motivation. Members from core-contributors will take a look (it might take some time - we are often overloaded with issues!) and discuss it. Once an agreement was reached - start coding.
-1. Start your work locally (usually until you need our CI testing).
+0. Think about what you want to do - fix a bug, repair docs, etc. If you want to implement a new feature or enhance an existing one.
+
+   - Start by opening a GitHub issue to explain the feature and the motivation.
+     In the case of features, ask yourself first - Is this NECESSARY for Lightning? There are some PRs which are just
+     purely about adding engineering complexity which have no place in Lightning.
+   - Core contributors will take a look (it might take some time - we are often overloaded with issues!) and discuss it.
+   - Once an agreement was reached - start coding.
+
+1. Start your work locally.
```

Comment:
```suggestion
0. Think about what you want to do - fix a bug, repair docs, etc. If you want to implement a new feature or enhance an existing one.

   - Start by opening a GitHub issue to explain the feature and the motivation.
     In the case of features, ask yourself first - Is this NECESSARY for Lightning? There are some PRs that are just
     purely about adding engineering complexity which has no place in Lightning.
   - Core contributors will take a look (it might take some time - we are often overloaded with issues!) and discuss it.
   - Once an agreement was reached - start coding.

1. Start your work locally.
```